crates/nargo/src/package.rs

```diff
@@ -1,19 +1,48 @@
-use std::{collections::BTreeMap, path::PathBuf};
+use std::{collections::BTreeMap, fmt::Display, path::PathBuf};
 
-use noirc_frontend::graph::{CrateName, CrateType};
+use noirc_frontend::graph::CrateName;
 
 use crate::constants::{PROVER_INPUT_FILE, VERIFIER_INPUT_FILE};
 
+#[derive(Debug, Copy, Clone)]
+pub enum PackageType {
+    Library,
+    Binary,
+}
+
+impl Display for PackageType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Library => write!(f, "lib"),
+            Self::Binary => write!(f, "bin"),
+        }
+    }
+}
+
 #[derive(Clone)]
 pub enum Dependency {
     Local { package: Package },
     Remote { package: Package },
 }
 
+impl Dependency {
+    pub fn is_binary(&self) -> bool {
+        match self {
+            Self::Local { package } | Self::Remote { package } => package.is_binary(),
+        }
+    }
+
+    pub fn package_name(&self) -> CrateName {
+        match self {
+            Self::Local { package } | Self::Remote { package } => package.name.clone(),
+        }
+    }
```

Comment:
```suggestion
    pub fn package_name(&self) -> &CrateName {
        match self {
            Self::Local { package } | Self::Remote { package } => package.name,
        }
    }
```

Let's delay cloning in case it's not necessary.

Reply:
Done. I figured it was a nicer API to clone since package name doesn't implement copy but it is a SmolStr so the clone is 0(1)